### ISSUE TITLE
EachWithObject Cop Incorrectly Flags Block

### ISSUE THREAD

**Issue by CyborgMaster** [rubocop/rubocop#1106]
I'm pretty sure I've found a bug in the EachWithObject cop.  It flags this call to `reduce` as one which should be changed to `each_with_object`

``` ruby
    used_replacement_variables.reduce(query) do |q, v|
      ReplacementVariableMap.query_for_variable v, q
    end
```

This uses ActiveRecord::Relation query objects and appends a bunch of query conditions into one query.  I do return a query object here, but each addition returns a new Relation object so it needs to be passed on to the next call to reduce.  

Maybe the cop just needs to be disabled with `# rubocop:disable` here, but it seems like this cop should be checking for a exact match on the return value.  For example if the methods looked like this:

``` ruby
    used_replacement_variables.reduce(query) do |q, v|
      ReplacementVariableMap.query_for_variable v, q
      q
    end
```

then I would expect it to be flagged.
